dataset_gen/src/camera_controls/scripts/driver_velocity_based.py

```diff
@@ -0,0 +1,154 @@
+                       model_state.pose.orientation.z,
+                       model_state.pose.orientation.w]
+
+        return position, orientation
+    except rospy.ServiceException as e:
+        print(f"Service call failed: {e}")
+
+
+# Изменение координат робота в соответствии с содержимым msg
+def coordinate_transformation(robot_pos, robot_dir):
+    # Величины, на которые изменятся координаты дрона
+    delta_pos = [x / 60 for x in [msg.x, msg.y, msg.z]]
+    # Величины, на которые изменятся углы дрона
+    delta_angle = quaternion_from_euler(*[x / 60 for x in [msg.roll, msg.pitch, msg.yaw]])
+
+    # Изменение координат дрона
+    delta_pos = rotate_by_quaternion(robot_dir, delta_pos)
+    robot_pos.x += delta_pos[0]
+    robot_pos.y += delta_pos[1]
+    robot_pos.z += delta_pos[2]
+
+    # Изменение углов дрона
+    robot_dir = quaternion_multiply(delta_angle, robot_dir)
+
+    return robot_pos, robot_dir
+
+
+# Телепортация дрона в глобальной системе координат
+def teleport_drone(robot_pos, robot_dir, drone_name):
+    rospy.wait_for_service('gazebo/set_model_state')
+    try:
+        sms = rospy.ServiceProxy('gazebo/set_model_state', SetModelState)
+        state = ModelState()
+        state.model_name = drone_name
+
+        state.pose.position = Point(robot_pos.x, robot_pos.y, robot_pos.z)
+
+        state.pose.orientation = Quaternion(*robot_dir)
+
+        return sms(state)
+    except rospy.ServiceException as e:
+        print(f"Service call failed: {e}")
+
+
+# Функция обработки сообщений
+def cmd_vel_event(data, drone_name):
+    msg.change_values(data)
+
+
+def listener():
+    # Если не указан параметр __name, то имя ноды по умолчанию -- 'driver'
+    rospy.init_node('driver')
+
+    # Имя дрона без слеша (e.g. 'drone1')
+    drone_name = rospy.get_name()[1:]
+
+    rospy.Subscriber(f'{drone_name}/cmd_vel', msg_transposition, cmd_vel_event, drone_name)
+
+    is_drone_prev = False
+    while not rospy.is_shutdown():
+        is_drone, err = is_drone_is_simulation_running(drone_name)
+        # Если симуляция запущена и дрон с именем drone_name есть на карте, то перемещаем его
+        if is_drone:
+            if is_drone and not is_drone_prev:
+                print(f'\'{drone_name}\' driver ready to receive messages')
+            try:
+                now = time.time()
+                robot_pos, robot_dir = get_drone_location(drone_name)
+                robot_pos, robot_dir = coordinate_transformation(robot_pos, robot_dir)
+                teleport_drone(robot_pos, robot_dir, drone_name)
+                delta_time = time.time() - now
+                rospy.sleep(1 / 60 - delta_time)
+            except (rospy.ROSInterruptException, TypeError):
+                rospy.signal_shutdown('Done')
+        else:
+            print(err)
+            time.sleep(1)
+        is_drone_prev = is_drone
+
+
+if __name__ == "__main__":
+    # Создание объекта сообщения
+    msg = Message()
```

Comment:
Weird syntax. Better declare instance in listener and make subscriber callback as lambda.
Or other way -- make it global variable and declare it before functions.